policy/api_spec_list.go

```diff
@@ -188,3 +188,62 @@ func (self *APISpecList) ReplaceHigherSharedSingleton(other *APISpecList) {
 		}
 	}
 }
+
+// For each microservice url, get the version range intersection among all occurances in the list.
+func (self *APISpecList) GetCommonVersionRanges() (*APISpecList, error) {
+	const NO_INTERSECTION = "NO_INTERSECTION"
+
+	new_list := new(APISpecList)
+
+	if len(*self) == 0 {
+		return new_list, nil
+	}
+
+	for _, apiSpec := range *self {
+		found := false
+		for i, newApiSpec := range *new_list {
+			if newApiSpec.SpecRef == apiSpec.SpecRef && newApiSpec.Org == apiSpec.Org && newApiSpec.Arch == apiSpec.Arch {
+				found = true
+
+				// ignore if previous has no intersection
+				if apiSpec.Version == NO_INTERSECTION {
```

Comment:
Good catch. It is redundant.  I must have thought about writing recursive function when I wrote that piece of code. I will remove it. Hope it does not break anything, I am scared of removing things because there were several times I tried to make something simple and later found out it was needed.

Reply:
Thats why we have unit tests, to catch us when we screw up. They catch me all the time.

Reply:
Done